### PR TITLE
nvme: Add Media Unit Status log page(LID: 0x10)

### PR DESCRIPTION
**Pull request by Arunpandian15**
Reviewed-by: Steven Seungcheol Lee <sc108.lee@samsung.com>
Signed-off-by: Arunpandian J <apj.arun@samsung.com>

### CODE REVIEW

File: Documentation/nvme-media-unit-stat-log.txt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+NAME
+----
+nvme-media-unit-stat-log - Send NVMe Media unit status Log pages request, returns result and log
+
+SYNOPSIS
+--------
+[verse]
+'nvme media-unit-stat-log' <device> [--dom-id | -d ]
+                            [--output-format=<fmt> | -o <fmt>]
+							[--raw-binary | -b]
```
Comment (igaw) [inline Documentation/nvme-media-unit-stat-log.txt:13]: line 12 uses spaces, line 13 tabs. please use consistent indention (see other docs what they are using here).

File: Documentation/nvme-media-unit-stat-log.txt [hunk 2]
```diff
@@ -0,0 +1,46 @@
+nvme-media-unit-stat-log(1)
+===========================
+
+NAME
+----
+nvme-media-unit-stat-log - Send NVMe Media unit status Log pages request, returns result and log
```
Comment (igaw) [inline Documentation/nvme-media-unit-stat-log.txt:6]: overlong line

File: Documentation/nvme-media-unit-stat-log.txt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+SYNOPSIS
+--------
+[verse]
+'nvme media-unit-stat-log' <device> [--dom-id | -d ]
+                            [--output-format=<fmt> | -o <fmt>]
+							[--raw-binary | -b]
+
+DESCRIPTION
+-----------
+Retrieves the NVMe Media unit status log pages details from an NVMe device and provides
```
Comment (igaw) [inline Documentation/nvme-media-unit-stat-log.txt:17]: overlong line

File: nvme-print.c [hunk 4]
```diff
@@ -2048,6 +2048,75 @@ void nvme_show_boot_part_log(void *bp_log, const char *devname,
...
+	}
+
+	json_object_add_value_array(root, "mus_list", entries);
+	json_print_object(root, NULL);
+	printf("\n");
+	json_free_object(root);
+
+}
+
+
```
Comment (igaw) [inline nvme-print.c:2086]: one newline too much

File: nvme-print.c [hunk 5]
```diff
@@ -2048,6 +2048,75 @@ void nvme_show_boot_part_log(void *bp_log, const char *devname,
...
+	int i;
+
+	root = json_create_object();
+	entries = json_create_array();
+
+	json_object_add_value_uint(root, "nmu", le16_to_cpu(mus->nmu));
+	json_object_add_value_uint(root, "cchans", le16_to_cpu(mus->cchans));
+	json_object_add_value_uint(root, "sel_config", le16_to_cpu(mus->sel_config));
+	for (i = 0; i < mus->nmu; i++) 
+	{
```
Comment (igaw) [inline nvme-print.c:2065]: the bracket goes to the line above

File: nvme-print.c [hunk 6]
```diff
@@ -2048,6 +2048,75 @@ void nvme_show_boot_part_log(void *bp_log, const char *devname,
...
+	struct json_object *entry;
+	int i;
+
+	root = json_create_object();
+	entries = json_create_array();
+
+	json_object_add_value_uint(root, "nmu", le16_to_cpu(mus->nmu));
+	json_object_add_value_uint(root, "cchans", le16_to_cpu(mus->cchans));
+	json_object_add_value_uint(root, "sel_config", le16_to_cpu(mus->sel_config));
+	for (i = 0; i < mus->nmu; i++)
```
Comment (igaw) [inline nvme-print.c:2064]: maybe add a new line between the json_object_add_value_uint() and the for () loop.

File: nvme-print.c [hunk 7]
```diff
@@ -2048,6 +2048,75 @@ void nvme_show_boot_part_log(void *bp_log, const char *devname,
...
+		json_object_add_value_uint(entry, "mucs", mus->mus_desc[i].mucs);
+		json_object_add_value_uint(entry, "cio", mus->mus_desc[i].cio);
+		json_array_add_value_object(entries, entry);
+	}
+
+	json_object_add_value_array(root, "mus_list", entries);
+	json_print_object(root, NULL);
+	printf("\n");
+	json_free_object(root);
+
```
Comment (igaw) [inline nvme-print.c:2083]: remove newline here

File: nvme-print.c [hunk 8]
```diff
@@ -2048,6 +2048,75 @@ void nvme_show_boot_part_log(void *bp_log, const char *devname,
...
+	if (flags & BINARY)
+		return d_raw((unsigned char *)mus_log, sizeof(*mus_log));
+	else if (flags & JSON)
+		return json_media_unit_stat_log(mus_log);
+
+	int nmu = le16_to_cpu(mus_log->nmu);
+	printf("Number of Media Unit Status Descriptors: %u\n", nmu);
+	printf("Number of Channels: %u\n", le16_to_cpu(mus_log->cchans));
+	printf("Selected Configuration: %u\n", le16_to_cpu(mus_log->sel_config));
+	int i;
```
Comment (igaw) [inline nvme-print.c:2099]: move the int declaration at the beginning of the function. Some people still compile this source with c89 standard.

File: nvme-print.c [hunk 9]
```diff
@@ -2048,6 +2048,75 @@ void nvme_show_boot_part_log(void *bp_log, const char *devname,
...
+	for (i = 0; i < nmu; i++) 
+	{
```
Comment (igaw) [inline nvme-print.c:2101]: bracket goes to the line above

File: nvme-print.c [hunk 10]
```diff
@@ -2048,6 +2048,75 @@ void nvme_show_boot_part_log(void *bp_log, const char *devname,
...
+
+void nvme_show_media_unit_stat_log(struct nvme_media_unit_stat_log *mus_log,
+				enum nvme_print_flags flags)
+{
+	if (flags & BINARY)
+		return d_raw((unsigned char *)mus_log, sizeof(*mus_log));
+	else if (flags & JSON)
+		return json_media_unit_stat_log(mus_log);
+
+	int nmu = le16_to_cpu(mus_log->nmu);
```
Comment (igaw) [inline nvme-print.c:2095]: Please move the int declaration at the beginning of the block. see next comment.

File: nvme.c [hunk 11]
```diff
@@ -1526,6 +1526,60 @@ static int get_boot_part_log(int argc, char **argv, struct command *cmd, struct
 	return nvme_status_to_errno(err, false);
 }
 
+static int get_media_unit_stat_log(int argc, char **argv, struct command *cmd,
+	struct plugin *plugin)
```
Comment (igaw) [inline nvme.c:1530]: Aling the line with the '('

**Review by igaw — CHANGES_REQUESTED** [review]
- Style nitpicks
- a couple of int deceleration at the start of the block (c89 support...)

File: Documentation/nvme-media-unit-stat-log.txt (hunk 1 above)
Comment (Arunpandian15) [reply Documentation/nvme-media-unit-stat-log.txt:13]: Addressed
Thanks @igaw

File: Documentation/nvme-media-unit-stat-log.txt (hunk 2 above)
Comment (Arunpandian15) [reply Documentation/nvme-media-unit-stat-log.txt:6]: Addressed
Thanks @igaw

File: Documentation/nvme-media-unit-stat-log.txt (hunk 3 above)
Comment (Arunpandian15) [reply Documentation/nvme-media-unit-stat-log.txt:17]: Addressed
Thanks @igaw

File: nvme-print.c (hunk 4 above)
Comment (Arunpandian15) [reply nvme-print.c:2086]: Addressed
Thanks @igaw

File: nvme-print.c (hunk 5 above)
Comment (Arunpandian15) [reply nvme-print.c:2065]: Addressed
Thanks @igaw

File: nvme-print.c (hunk 7 above)
Comment (Arunpandian15) [reply nvme-print.c:2083]: Addressed
Thanks @igaw

File: nvme-print.c (hunk 6 above)
Comment (Arunpandian15) [reply nvme-print.c:2064]: Addressed
Thanks @igaw

File: nvme-print.c (hunk 8 above)
Comment (Arunpandian15) [reply nvme-print.c:2099]: Addressed
Thanks @igaw

File: nvme-print.c (hunk 9 above)
Comment (Arunpandian15) [reply nvme-print.c:2101]: Addressed
Thanks @igaw

File: nvme-print.c (hunk 10 above)
Comment (Arunpandian15) [reply nvme-print.c:2095]: Addressed
Thanks @igaw

File: nvme.c (hunk 11 above)
Comment (Arunpandian15) [reply nvme.c:1530]: Addressed
Thanks @igaw

**Comment by igaw** [comment]
Depends on https://github.com/linux-nvme/libnvme/pull/157

**Comment by Arunpandian15** [comment]
Closing this PR and will raise New PR since this branch has not aligned with recent change in nvme-cli

**Pull request closed without merging**